doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+   requires the time to be accurate within 5-10 minutes.
+
+6. **Sensor data**. The data received from the sensor itself, and
+   delivered to the service shall be delivered without modification or
+   tampering.
+
+7. **Device configuration**. Various configuration data, such as the
+   hostname of the service to connect to, the address of a time server,
+   frequency and parameters of when sensor data is sent to the service
+   and other need to be kept by the device. This configuration data will
+   need to be updated periodically as the configuration changes. Updates
+   should be allowed only from authorized parties.
+
+8. **Logs**. In order to assist with analysis of security issues, the
+   device shall log information about security-pertinent events. IoT
+   devices generally have limited storage, and as such, these logs need
+   to be carefully selected. It may also be possible to send these log
+   events to the cloud service where they can be stored in a more
+   resource-available environment. Types of events that should be logged
+   include:
+
+   a. **Firmware image updates**. The system should log the download of
+      new images, and when an image is successfully updated.
+
+   b. **Client secret changes**. And new client secrets should be
+      logged.
+
+   c. **Changes to the device configuration**.
+
+Communication
+=============
+
+In addition to assets, the threat model also considers the locations
+where data or assets are communicated between entities of the system.
+
+1. **Flash contents**. The flash device contains several regions. The
+   contents of flash can be modified programmatically by the SoC’s CPU.
+
+   a. **The bootloader**. As described in the Assets section, the
+      bootloader is a small section of the flash device containing the
+      code initially run. This section shall be written early in the
+      lifecycle of the device, and the flash device then configured to
+      permanently disallow modification of this section. This
+      configuration should also prevent modification via external
+      interfaces, such as JTAG or SWD debuggers.
+
+      The bootloader is responsible for verifying the signature of the
+      application image as well as updating the application image from
+      the update image when an update is needed.
+
+      The bootloader shall verify the signature of the update image
+      before installing it.
+
+      The bootloader shall only accept an update image with a newer
+      version number than the current image.
+
+   b. **The application image**. The application image contains the code
+      executed during normal operation of the device. Before running
+      this image, the bootloader shall verify a digital signature of the
+      image, to avoid running an image that has been tampered with. The
+      flash/system shall be configured such that after the bootloader
+      has completed, the CPU will be unable to write to the application
+      image.
```

Comment:
@d3zd3z is MCUBoot currently aligned with this?

Reply:
Not currently, no.